Library/Homebrew/livecheck/strategy/github_release.rb

```diff
@@ -0,0 +1,136 @@
+        DEFAULT_REGEX = /v?(\d+(?:\.\d+)+)/i.freeze
+
+        # Keys in the JSON that could contain the version.
+        VERSION_KEYS = ["name", "tag_name"].freeze
+
+        # Whether the strategy can be applied to the provided URL.
+        #
+        # @param url [String] the URL to match against
+        # @return [Boolean]
+        sig { params(url: String).returns(T::Boolean) }
+        def self.match?(url)
+          URL_MATCH_REGEX.match?(url)
+        end
+
+        # Uses the regex to match text in the content or, if a block is
+        # provided, passes the page content to the block to handle matching.
+        # With either approach, an array of unique matches is returned.
+        #
+        # @param content [String] the page content to check
+        # @param regex [Regexp, nil] a regex used for matching versions in the
+        #   content
+        # @return [Array]
+        sig {
+          params(
+            content: T::Array[T::Hash[String, T.untyped]],
+            regex:   T.nilable(Regexp),
+            block:   T.nilable(Proc),
+          ).returns(T::Array[String])
+        }
+        def self.versions_from_content(content, regex, &block)
+          if block
+            block_return_value = regex.present? ? yield(content, regex) : yield(content)
+            return Strategy.handle_block_return(block_return_value)
+          end
+
+          values = []
+          content.each do |release|
+            next if release.blank?
+
+            VERSION_KEYS.each do |key|
+              match = release[key].match(regex)
+
+              next if match.blank?
+
+              values.push(match.to_s)
+              break
+            end
+          end
+
+          values.compact.uniq
+        end
+
+        # Generates a URL and regex (if one isn't provided) and passes them
+        # to {PageMatch.find_versions} to identify versions in the content.
+        #
+        # @param url [String] the URL of the content to check
+        # @param regex [Regexp] a regex used for matching versions in content
+        # @return [Hash]
+        sig {
+          params(
+            url:     String,
+            regex:   T.nilable(Regexp),
+            _unused: T.nilable(T::Hash[Symbol, T.untyped]),
+            block:   T.nilable(Proc),
+          ).returns(T::Hash[Symbol, T.untyped])
+        }
+        def self.find_versions(url:, regex: nil, **_unused, &block)
+          match_data = { matches: {}, regex: regex || GithubRelease::DEFAULT_REGEX }
+          match = url.sub(/\.git$/i, "").match(URL_MATCH_REGEX)
+
+          return match_data if match.blank?
+
+          releases = GitHub::API.open_rest("https://api.github.com/repos/#{match[:username]}/#{match[:repository]}/releases")
+
+          GithubRelease.versions_from_content(releases, regex || DEFAULT_REGEX, &block).each do |match_text|
+            match_data[:matches][match_text] = Version.new(match_text)
+          end
+
+          match_data
+        end
+      end
```

Comment:
```suggestion
      end
      GitHubRelease = GithubRelease 
```

because I'm a loser and then `GitHubRelease` can be used everywhere which reads nicer

Reply:
We should just call it `GitHubRelease` from the beginning tbh.

Also, not sure why we'd ever use `GithubLatest` if we had this.

Reply:
I think it's called `GithubRelease` so it's mapped to `:github_release` instead of `:git_hub_release`.

Reply:
I mean we can just name it correctly and not do the mapping based on the class name.

Reply:
By adding this alias I think it will actually allow both `:github_release` and `:git_hub_release` to work, which we don't want.

Reply:
If we're really against the current class name format (for the sake of the generated symbol), we could just add a constant for the symbol to each strategy. I considered this a while back (2020) when creating the current strategy structure but went with a programmatic approach for the sake of simplicity. Some may argue that having to use a specific class name format is _not_ simple and I don't disagree but changing this would break any third-party tap strategies (i.e., they wouldn't be recognized by livecheck) until they're updated.

Regardless of how it's handled, it needs to be something that third-party taps can do in their own strategies (i.e., we can't just hardcode first-party strategies in `Strategy#strategies` and call it a day).

That said, it's easier if we simply use `GithubRelease` for now and tackle this issue in a separate PR (if we pursue it). I agree that we don't want something like `:git_hub_release` in the mix.

Reply:
> we can't just hardcode first-party strategies in Strategy#strategies and call it a day

Of course we can. 😄 I wasn't even aware we support third-party strategies. In my opinion our strategies should be enough for pretty much any third-party.

If third-parties need to do anything more advanced than what we support, they can just use `bump-*` somewhere in their release pipeline.

> That said, it's easier if we simply use `GithubRelease` for now

Yes, definitely should be done in a separate PR.

Reply:
> Of course we can. 😄 I wasn't even aware we support third-party strategies. In my opinion our strategies should be enough for pretty much any third-party.
> 
> If third-parties need to do anything more advanced than what we support, they can just use `bump-*` somewhere in their release pipeline.

[We've quietly supported tap strategies for years](Homebrew/homebrew-livecheck/pull/1025) and I've kept it in mind when making changes since then. Fwiw, `Xorg` was originally a tap strategy (from Maxim's homebrew-xorg tap) before we internalized it (I reworked the strategy in the process).

Every now and then someone creates an issue and references their tap strategies, so I don't have any intention of dropping support without a really good reason. It's fine if third-party taps have to make some small changes to their own strategies but honestly we should be _reducing_ the areas where we're referencing specific strategies in livecheck (I have some stashed work in this vein that involves moving certain logic into strategies).

Reply:
The alias approach is preferable IMO because `github_release` is the "expected" filename based on the class name. It's what I do in Rails so the autoloader can find it.